neuralmonkey/decoders/beam_search_decoder.py

```diff
@@ -94,6 +96,19 @@ def __init__(self,
         self._beam_size = beam_size
         self._length_normalization = length_normalization
 
+        if (hasattr(self.parent_decoder, "encoder_states")
+                and self.parent_decoder.encoder_states is not None):
+            enc_states = self.parent_decoder.encoder_states
+            enc_states_shape = tf.shape(enc_states)
+            self.parent_decoder.encoder_states = tf.reshape(
```

Comment:
Tohle přepisuje parent decoder, to by neměl, ne? To by znamenalo, že jakmile se vytvoří `BeamSearchDecoder`, tak už nepůjde pustit greedy ani trénování.

Reply:
No jedine dalsi reseni, co me napada udelat duplikat encoder_states ve feedables a attendovat nad temito feedables.encoder_states